### ISSUE TITLE
Invalid filename in shebang line

### ISSUE THREAD

**Issue by connorlanigan** [hqjs/hq#9]
When I run `hq`, I always get an error. This error varies slightly, depending on the version of `coreutils` (which contains the `env` executable).

Ubuntu 18.04.3 (LTS) (coreutils v8.28):
```
~ $ hq
/usr/bin/env: ‘node --experimental-modules --no-warnings’: No such file or directory
```

Ubuntu 19.04 (coreutils v8.30):
```
~ $ hq
/usr/bin/env: 'node --experimental-modules --no-warnings': No such file or directory
/usr/bin/env: use -[v]S to pass options in shebang lines
```

I'm using hq version 0.0.12.

This seems to be caused by the shebang in this file: https://github.com/hqjs/hq/blob/1cd20a0a688c7468e357cacc0ef9ea3089746963/index.mjs#L1

From what I found out so far, a shebang can only contain a single parameter. In this case, the string `node --experimental-modules --no-warnings` is passed as a whole to `/usr/bin/env`, and since this is not an existing file, it cannot be executed.

A possible solution (as suggested in the second error message) is to use the `-S` flag, which splits up the argument at the whitespaces. That would look like this:
```
#!/usr/bin/env -S node --experimental-modules --no-warnings
```

(Unfortunately, this flag was only introduced in version 8.30 of coreutils, and as such is not present in older Ubuntu versions. However - if I understand correctly - the current version of the script doesn't work on those versions anyway, so there is nothing to lose there.)

Can you confirm if this script works on other Linux versions or on Mac?

**Comment by hqjs** [comment]
Hi @connorlanigan. Thank you for the report and PR. Both original and modified scripts are working on mac. I think it might be better to remove second option `--no-warnings` rather then break compatibility with older coreutils (as a result it will only complains about ESM module loader that will stop being experimental in October).

Any chances that it will work with windows as well? May it be better to keep `--no-wornings`, but remove shabeng and change `start` and `bin` scripts in `package.json`?

**Comment by connorlanigan** [comment]
Removing the `--no-warnings` flag would not change anything without also removing `--experimental-modules`, since we can only pass a single argument _to /usr/bin/env_. We'd have to only use "node" as the single argument to /usr/bin/env in order to make it find the right executable. But without the `experimental-modules` flag, the tool wouldn't work, right?

# 
> rather then break compatibility with older coreutils

Currently, I think there is no compatibility at all under Linux (I've only tried Ubuntu, though I guess most distros use the same /usr/bin/env). The `-S`-fix would fix this for those Linux distros that are using the newer coreutils, while older ones would remain unchangedly non-working (but over time, they'll disappear). This would be the easiest way and what I'd recommend.

# 
Removing the shebang alone won't work, since then Linux can't execute the file anymore. It might be possible to create a shell script instead which then only contains a call to run the actual hq script. Something like:
```
#!/bin/bash
/usr/bin/env node --experimental-modules --no-warnings <path-to-index.mjs> "$@"
```
From what I researched so far, though, it seems extremely complicated to reliably specify the path to `index.mjs` so that it can be called from anywhere.

# 
I think that Windows simply executes the file by its extension, i.e. it runs whatever program is associated with *.mjs files. I have no idea how this affects Windows and I don't have a Windows machine to test it on.

**Comment by hqjs** [comment]
I mean after removing shabeng it is possible to change `package.json` this way:
```json
"scripts": {
    "dev": "nodemon --experimental-modules index.mjs",
    "start": "node --experimantal-modules --no-warnings index.mjs"
  },
  "bin": {
    "hq": "node --experimantal-modules --no-warnings index.mjs"
  },
```
and it will assure compatibility for linux, mac and windows (did not work with windows for many years so not 100% sure about that)

**Comment by hqjs** [comment]
seems like it is not working as `bin` requires a script or binary, not a command.

**Comment by connorlanigan** [comment]
Yes, the `bin` entries are used for creating symlinks, so they can only be files. This is what my `bin/` folder looks like:
```
gh -> ../lib/node_modules/gh/bin/gh.js*
hq -> '../lib/node_modules/@hqjs/hq/index.mjs'*
hs -> ../lib/node_modules/http-server/bin/http-server*
http-server -> ../lib/node_modules/http-server/bin/http-server*
netlify -> ../lib/node_modules/netlify-cli/bin/run*
ntl -> ../lib/node_modules/netlify-cli/bin/run*
parcel -> ../lib/node_modules/parcel/bin/cli.js*
serve -> ../lib/node_modules/serve/bin/serve.js*
tsc -> ../lib/node_modules/typescript/bin/tsc*
ts-node -> ../lib/node_modules/ts-node/dist/bin.js*
tsserver -> ../lib/node_modules/typescript/bin/tsserver*
wrangler -> '../lib/node_modules/@cloudflare/wrangler/run-wrangler.js'
```

**Comment by hqjs** [comment]
This script will work, but for UNIX only
```bash
#!/bin/bash
node_root=`dirname $0`
rel_path=`readlink $0`
script_path=`dirname $node_root/$rel_path`

/usr/bin/env node --experimental-modules --no-warnings $script_path/index.mjs
```

**Comment by zerdos** [comment]
same error for me as well

**Comment by hqjs** [comment]
@zerdos Can you please specify some details about your system?
* OS
* coreutils version
* nodejs version
* hq version

**Comment by zerdos** [comment]
Tried with Ubuntu bionic with node 12, node 13, 
Eoan node 12, 
Debian Buster node 12, node 13
Alpine node 12 , node 13,
And windows 10, node 13.

Nothing worked.

**Comment by zerdos** [comment]
(Also with the official node docker images, you couln't even install the package with npm, only just yarn - but after the installation its still having this env issue)

**Comment by hqjs** [comment]
On mac it used to work since node 8.9, but now they have remove `--experimental-modules` option and add `--experimental-json-module` instead. I will consider to migrate to commonjs or use starter script to avoid this kind of problems.

For now I would suggest to run it manually `node --experimental-modules --no-warnings /path/to/hq/installation/index.mjs` for node < 12.13 or `node --experimental-json-modules --no-warnings /path/to/hq/installation/index.mjs`.

Sorry for inconvenience.

**Comment by hqjs** [comment]
@zerdos please check @hqjs/hq@0.0.17 the problem should be fixed
